example/src/Examples/API/AnimatedImages.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect } from "react";
+import { ScrollView, useWindowDimensions } from "react-native";
+import {
+  Canvas,
+  Image,
+  SkAnimatedImage,
+  SkImage,
+  useAnimatedImage,
+} from "@shopify/react-native-skia";
+import { useSharedValue } from "react-native-reanimated";
+import { base64Gif } from "./base64Gif";
+
+export const AnimatedImages = () => {
+  const { width: wWidth } = useWindowDimensions();
+  const SIZE = wWidth / 3;
+  const S2 = 60;
+  const PAD = (SIZE - S2) / 2;
+
+  // Verifies that the error handler for animated images are working correctly.
+  useAnimatedImage(new Uint8Array([0, 0, 0, 255]), (err) => {
+    if (err.message !== "Could not load data") {
+      throw new Error(
+        `Expected error message to be 'Could not load data' - got '${err.message}'`
+      );
+    }
+  });
+  useAnimatedImage("https://reactjs.org/invalid.jpg", (err) => {
+    if (err.message !== "Could not load data") {
+      throw new Error(
+        `Expected error message to be 'Could not load data' - got '${err.message}'`
+      );
+    }
+  });
+
+  // Verifies that we can use this hook with a null/undefined input parameter
+  useAnimatedImage(null);
+  useAnimatedImage(undefined);
+
+  const example1 = useAnimatedImage(require("../../assets/birdFlying.gif"));
+  const currentFrame1 = useCurrentFrame(example1);
+
+  const example2 = useAnimatedImage(base64Gif);
+  const currentFrame2 = useCurrentFrame(example2);
+
+  return (
+    <ScrollView>
+      <Canvas
+        style={{
+          alignSelf: "center",
+          width: 320,
+          height: 180,
+          marginVertical: PAD,
+        }}
+      >
+        <Image
+          image={currentFrame1}
+          x={0}
+          y={0}
+          width={320}
+          height={180}
+          fit="contain"
+        />
+      </Canvas>
+      <Canvas
+        style={{
+          alignSelf: "center",
+          width: 320,
+          height: 180,
+          borderColor: "#aaaaaa",
+          borderWidth: 1,
+          borderStyle: "solid",
+          marginVertical: PAD,
+        }}
+      >
+        <Image
+          image={currentFrame2}
+          x={0}
+          y={0}
+          width={320}
+          height={180}
+          fit="contain"
+        />
+      </Canvas>
+    </ScrollView>
+  );
+};
+
+/**
+ * Custom hook that returns the current frame of an animated image
+ *
+ * @param animatedImage
+ * @returns the current frame as a SkImage or null
+ */
+const useCurrentFrame = (animatedImage: SkAnimatedImage | null) => {
```

Comment:
haha this works surprisingly well :) 
But could we remove it and use Reanimated instead?

Reply:
@wcandillon I removed this hook and updated the example to use reanimated (`useFrameCallback`). It's working great on ios and web - I'm still seeing the frame drops on android so that will need to be investigated.